pkg/zbm/reader.go

```diff
@@ -75,20 +64,20 @@ func Decode(r io.Reader) (image.Image, error) {
 
 	pixelBuffer := make([]uint16, c.width*c.height)
 
-	// swap every two pixels
-	// TODO why does it look better that way?
+	// swap every two pixels, endianness changes a bit
 	for i := 0; i < len(pixelBuffer)-1; i += 2 {
 		pixelBuffer[i+1] = binary.BigEndian.Uint16(buffer[i*2 : (i*2)+2])
-		pixelBuffer[i] = binary.BigEndian.Uint16(buffer[i*2 : (i*2)+2])
+		pixelBuffer[i] = binary.BigEndian.Uint16(buffer[(i+1)*2 : ((i+1)*2)+2])
 	}
 
 	img := image.NewRGBA(image.Rect(0, 0, int(c.width), int(c.height)))
-	//pixels := img.Pix[0 : int(c.width)*int(c.height)*4]
+
 	for i := 0; i < int(c.width*c.height); i++ {
-		r, g, b, a := getPixelValue(pixelBuffer[i])
-		img.Pix[4*i] = r
+		cb, cr, y, a := getPixelValue(pixelBuffer[i])
```

Comment:
Is this a typo `cb, cr, y, a`, since `getPixelValue` returns them in `cr, cb, y, alpha` order? That might explain why your output looks more blue and less green on the lower panel.

Reply:
Thank you, this was a typo :)

Reply:
Welcome. If you find any more novel formats in this Game Wave research, I am eager to catalog them (I have a few hundred so far from various video game consoles).